src/App/Fossa/Config/Analyze.hs

```diff
@@ -273,6 +290,15 @@ cliParser =
     <*> monorepoOpts
     <*> baseDirArg
 
+vendoredDependencyModeOpt :: Parser ArchiveUploadType
+vendoredDependencyModeOpt = option (maybeReader parseType) (long "force-vendored-dependency-scan-method" <> metavar "METHOD" <> help "Force the vendored dependency scan method. The options are 'CLILicenseScan' or 'ArchiveUpload'. 'CLILicenseScan' is usually the default unless your organization has overridden this.")
+  where
+    parseType :: String -> Maybe ArchiveUploadType
+    parseType = \case
+      "ArchiveUpload" -> Just ArchiveUpload
+      "CLILicenseScan" -> Just CLILicenseScan
+      _ -> Nothing
```

Comment:
I think this means that if I pass in an arbitrary string to `force-vendored-dependency-scan-method`, it'll just use the default. Is that correct? (I don't see anything that causes the `Nothing` case to error if they provided a value)

Basically I'm looking for something like this in this changeset: https://github.com/fossas/fossa-cli/blob/102a09e1919379868739872e4939aff13529bf95/src/App/Fossa/Config/Report.hs#L192-L195

Reply:
It does actually fail, but with a not-so-great error message:

```
option --force-vendored-dependency-scan-method: cannot parse value `CLILicenseScaaaan'
```

I've fixed it by using `eitherReader` instead of `maybeReader`. This makes it more obvious that there's an error message, and gives you the possibility of a better error message too. The error message now looks like this:

```
option --force-vendored-dependency-scan-method: must be either 'CLILicenseScan' or 'ArchiveUpload'. Found CLILicenseScaaaan
```

Reply:
Brilliant, thanks! ✨